editoast/src/main.rs

```diff
@@ -164,3 +171,55 @@ fn generate(
     }
     Ok(())
 }
+
+#[derive(Debug)]
+struct InfraNameAlreadyExistsError;
+
+impl Error for InfraNameAlreadyExistsError {}
+
+impl Display for InfraNameAlreadyExistsError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Infra name already exists")
+    }
+}
```

Comment:
## Consistency

So far, we allow to have infrastructures with the same name. It's a bit weird to trigger an error only here. It might be a good idea to prohibit infrastructures with the same name but that should be done in another PR.

## Rust improvements

You could make an enum `ImportRailjsonError` and use [thiserror](https://docs.rs/thiserror/latest/thiserror/) to make it an error.

We could move this error and the `import_railjson` function to another dedicated file.

Reply:
If you follow the first point you should remove `InfraNameAlreadyExistsError` so you don't have to move it.